src/Credit/Client.php

```diff
@@ -0,0 +1,181 @@
+    protected const TAG_BILL = 'bill';
+    protected const TAG_PAY_DATE = 'payDate';
+    protected const TAG_PAY_AMOUNT = 'paidAmount';
+    protected const TAG_STATUS = 'status';
+    protected const TAG_PAYEE = 'payee';
+
+    protected const ATTRIBUTE_CODE = 'code';
+    protected const ATTRIBUTE_ID = 'id';
+
+    /** @var Config */
+    protected $config;
+
+    /** @var GuzzleHttp\Client */
+    protected $guzzleClient;
+
+    /** @var LuhnAlgorithm\Contract\LuhnAlgorithmInterface */
+    protected $luhn;
+
+    /**
+     * Client constructor.
+     *
+     * @param Config            $config
+     * @param GuzzleHttp\Client $guzzleClient
+     */
+    public function __construct(Config $config, GuzzleHttp\Client $guzzleClient)
+    {
+        $this->config = $config;
+        $this->guzzleClient = $guzzleClient;
+        $this->luhn = LuhnAlgorithm\LuhnAlgorithmFactory::create();
+    }
+
+    /**
+     * @param Credit\TransferInterface $creditToCard
+     *
+     * @return Credit\Response
+     * @throws Credit\Exception
+     * @throws GuzzleHttp\Exception\GuzzleException
+     */
+    public function send(Credit\TransferInterface $creditToCard): Credit\Response
+    {
+        $this->validateCardNumber($creditToCard);
+
+        $response = $this->guzzleClient->request('post', $this->config->getUrl(), [
+            'login' => $this->config->getKey(),
+            'password' => $this->config->getSecret(),
+            'method' => static::BILL_CREATE,
+            'payeeId' => $this->config->getPayee(),
+            'contractNumber' => $creditToCard->getCardToken()
+        ]);
+
+        $this->validateTransfer($creditToCard, $response);
+
+        $document = simplexml_load_string(
+            $response->getBody()->__toString()
+        );
+
+        $response = $this->guzzleClient->request('post', $this->config->getUrl(), [
+            'login' => $this->config->getKey(),
+            'password' => $this->config->getSecret(),
+            'method' => static::BILL_PAY,
+            'billId' => $document->{static::TAG_BILL}->attributes()[static::ATTRIBUTE_ID],
+            'amount' => $creditToCard->getAmount()
+        ]);
+
+        $this->validateTransfer($creditToCard, $response);
+
+        $response = simplexml_load_string(
+            $response->getBody()->__toString()
+        )->{static::TAG_BILL};
+
+        return new Payment(
+            $response->attributes()[static::ATTRIBUTE_ID],
+            Carbon::createFromFormat('d.m.Y H:i:s', (string)$response->{static::TAG_PAY_DATE}),
+            (float)$response->{static::TAG_PAY_AMOUNT},
+            (string)$response->{static::TAG_STATUS}
+        );
+    }
+
+    /**
+     * @param Credit\TransferInterface $creditToCard
+     *
+     * @throws Credit\Exception
+     */
+    protected function validateCardNumber(Credit\TransferInterface $creditToCard): void
+    {
+        $card = $creditToCard->getCardToken();
+
+        if (!preg_match('/\d{16}/', $card)
```

Comment:
Incorrect card checking. 
Card number `a1234123412341234b` will pass this regexp